tests/apps/stateapp/app.go

```diff
@@ -353,30 +414,54 @@ func httpHandler(w http.ResponseWriter, r *http.Request) {
 	statestore := mux.Vars(r)["statestore"]
 	switch cmd {
 	case "save":
-		err = save(req.States, statestore)
+		if req, err = parseRequestBody(w, r); err != nil {
+			return
+		}
+		_, err = save(req.States, statestore)
 		if err == nil {
 			// The save call to dapr side car has returned correct status.
 			// Set the status code to statusNoContent
 			statusCode = http.StatusNoContent
 		}
+	case "load":
+		if data, err = getRequestBody(w, r); err != nil {
+			return
+		}
+		statusCode, err = load(data, statestore)
 	case "get":
-		states, err = getAll(req.States, statestore)
-		res.States = states
+		if req, err = parseRequestBody(w, r); err != nil {
+			return
+		}
+		res.States, err = getAll(req.States, statestore)
 	case "getbulk":
-		states, err = getBulk(req.States, statestore)
-		res.States = states
+		if req, err = parseRequestBody(w, r); err != nil {
+			return
+		}
+		res.States, err = getBulk(req.States, statestore)
 	case "delete":
+		if req, err = parseRequestBody(w, r); err != nil {
+			return
+		}
 		err = deleteAll(req.States, statestore)
 		statusCode = http.StatusNoContent
 	case "transact":
+		if req, err = parseRequestBody(w, r); err != nil {
+			return
+		}
 		err = executeTransaction(req.States, statestore)
+	case "query":
+		if data, err = getRequestBody(w, r); err != nil {
+			return
```

Comment:
same here.

Reply:
the error is handled inside `getRequestBody`